function-template/Dockerfile

```diff
@@ -1,11 +1,17 @@
 ARG IMAGE
 FROM ${IMAGE}
 
-ARG FUNCTION_SRC=function.txt
+ARG HANDLER
+ENV HANDLER=$HANDLER
 
-# FUNCTION_MODULE env var is set in the base-image
-COPY ${FUNCTION_SRC} ${FUNCTION_MODULE}
+COPY . .
 
-RUN echo "  (require '[func-server.main :as m]) (m/run-server function)" >> ${FUNCTION_MODULE}
+RUN echo -n ${HANDLER} > /tmp/handler && \
+    [ -z "${HANDLER}" ] && \
+    ([ $(ls *.clj | wc -l) -eq "1" ] || (echo "Exactly 1 source file needed when HANDLER is empty" >&2 && exit 1)) && \
+    (echo $(clojure -m func-server.read-ns $(ls *.clj))/handle > /tmp/handler || \
+      (echo "Could not parse the namespace from source file $(ls *.clj)" >&2 && exit 1)) && \
+    ([ $(clojure -i $(ls *.clj) -e "(fn? $(</tmp/handler))") == "true" ] || \
+      (echo "Could not find 'handle' fn in source file $(ls *.clj)" >&2 && exit 1))
 
-CMD clojure ${FUNCTION_MODULE}
+CMD clojure $([ -z "${HANDLER}" ] && echo "-i" "$(ls *.clj)") -m func-server.main $(</tmp/handler)
```

Comment:
Does this support single files with a handler arg? If a user wants to specify a handler arg with a single file, it should still work.